locksmith/src/operations/generate-pass/ios/createAppleWalletPass.ts

```diff
@@ -13,15 +13,37 @@ const modelPath = path.resolve(
 // Retrieve certificates necessary for signing the pass
 const { wwdr, signerCert, signerKey, signerKeyPassphrase } = getCertificates()
 
+// Utility to map content types to file extensions
+const contentTypeToExtensionMap: { [key: string]: string } = {
+  'image/png': 'png',
+  'image/jpeg': 'jpg',
+  'image/jpg': 'jpg',
+  'image/gif': 'gif',
+  'image/webp': 'webp',
+  'image/svg+xml': 'svg',
+}
+
 // utility to retrieve an image via HTTP and return it as a Buffer
 async function fetchImageAsBuffer(imageUrl: string) {
   const response = await fetch(imageUrl)
   if (!response.ok) {
     throw new Error(`Failed to fetch image: ${response.statusText}`)
   }
+  const contentType = response.headers.get('content-type')
+  const extension = contentType
+    ? contentTypeToExtensionMap[contentType] || 'png'
+    : 'png'
   const imageArrayBuffer = await response.arrayBuffer()
   // Convert the ArrayBuffer to a Node.js Buffer
-  return Buffer.from(imageArrayBuffer)
+  const buffer = Buffer.from(imageArrayBuffer)
+  return { buffer, extension }
+}
+
+// utility to generate serial numbers
+function generateRandomSerialNumber() {
```

Comment:
Hum, can you explain this a bit please?

Reply:
sure!

this generates a serial number for every generated pass, as it is required by apple